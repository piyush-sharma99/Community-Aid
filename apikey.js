/*
 *  ClassName: apikey.js
 *
 *  Date: 28/03/2021
 *
 * @author Piyush Sharma, X17342356
 *
 * @reference https://reactnative.dev/docs/activityindicator
 * @reference https://www.udemy.com/course/react-native-the-practical-guide/
 * @reference https://www.youtube.com/watch?v=8zT6CYu0iYQ
 *
 */

export const firebaseConfig = {
  apiKey: "AIzaSyDGliQfMfKpXUllG5ZPyClJL6C8VgFKXo8",
  authDomain: "community-aid.firebaseapp.com",
  databaseURL:
    "https://community-aid-default-rtdb.europe-west1.firebasedatabase.app",
  projectId: "community-aid",
  storageBucket: "community-aid.appspot.com",
  messagingSenderId: "529896701339",
  appId: "1:529896701339:web:1c30369d39f8654b01353e",
  measurementId: "G-PPD28P0PC6",
};
