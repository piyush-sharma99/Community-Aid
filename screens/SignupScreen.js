import React, {useState} from 'react';
import { StyleSheet, Text, View, TextInput, Button, TouchableOpacity, Image, } from 'react-native';
import * as firebase from 'firebase';
import signUpUser from '../functions/signUpUser';
import { KeyboardAwareScrollView } from 'react-native-keyboard-aware-scroll-view';

const SignupScreen = props => {

  const db = firebase.firestore();
  const [name, setName] = useState('');
  const [number, setNumber] = useState('');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [passwordRepeat, setPasswordRepeat] = useState('');
  const fb = firebase.auth();

  resetFieldsOne = () => {
    this.textInputOne.clear();     
    this.textInputTwo.clear(); 
    this.textInputThree.clear(); 
    this.textInputFour.clear(); 
    this.textInputFive.clear();  
}


  return(
    <KeyboardAwareScrollView style={styles.container}>
    <View style={styles.screen} >
        
    <Image 
    style={styles.logo}
    source={require('../assets/Logo.png')}
    />

    <View style={styles.inputView} >
      <TextInput  
        style={styles.inputText}
        placeholder="Name..." 
        placeholderTextColor="#003f5c"
        onChangeText={(name) => setName(name)}
        ref={input => { this.textInputOne = input }}/>
    </View>

    <View style={styles.inputView} >
      <TextInput  
        style={styles.inputText}
        placeholder="Number..." 
        placeholderTextColor="#003f5c"
        onChangeText={(number) => setNumber(number)}
        ref={input => { this.textInputTwo = input }}/>
    </View>

     <View style={styles.inputView} >
      <TextInput  
        style={styles.inputText}
        placeholder="Email..." 
        placeholderTextColor="#003f5c"
        onChangeText={(email) => setEmail(email)}
        ref={input => { this.textInputThree = input }}/>
    </View>

    <View style={styles.inputView} >
      <TextInput  
        style={styles.inputText}
        placeholder="Password..." 
        secureTextEntry
        placeholderTextColor="#003f5c"
        onChangeText={(password) => setPassword(password)}
        ref={input => { this.textInputFour = input }}/>
    </View>
    <View style={styles.inputView} >
      <TextInput  
        style={styles.inputText}
        placeholder="Repeat Password..." 
        secureTextEntry
        placeholderTextColor="#003f5c"
        onChangeText={(passwordRepeat) => setPasswordRepeat(passwordRepeat)}
        ref={input => { this.textInputFive = input }}/>
    </View>

    <View>
    <TouchableOpacity style={styles.signBtn}  onPress = {() => signUpUser(name, number, email, password, passwordRepeat, fb, props, db)} onPressOut = {resetFieldsOne}>
      <Text style={styles.signText}>Sign Up</Text>
    </TouchableOpacity>
    </View>

    </View>
    </KeyboardAwareScrollView>
   
);

};

SignupScreen.navigationOptions = {
headerTitle: 'Sign up',
headerStyle: {
backgroundColor: '#2E86C1'
},
headerTintColor:"white"
};

const styles = StyleSheet.create({
    screen: {
        width:'100%',
        height:'100%',
        flex: 1,
        backgroundColor: '#2E86C1',
        alignItems: 'center',
        justifyContent: 'center'
    },

    container: {
      backgroundColor: '#2E86C1',
    
  },

    logo:{
        justifyContent:"center",
        marginTop:-20
      },

      inputView:{
        width:"80%",
        backgroundColor:"#FDFEFE",
        borderRadius:25,
        height:50,
        marginTop:20,
        marginBottom:20,
        justifyContent:"center",
        padding:20,
        elevation: 10,
        borderWidth: 4,
        borderColor: "#fb5b5a"
      },

      inputText:{
        height:50,
        color:"black"
      },

      signText:{
        color:"white",
        fontSize:20
      },


      signBtn:{
        width:200,
        backgroundColor:"#fb5b5a",
        borderRadius:25,
        height:50,
        alignItems:"center",
        justifyContent:"center",
        borderWidth: 3,
        borderColor: '#fff',
        borderRadius:15,
        shadowOpacity: 1,
        elevation:10,
        marginTop:50,
        marginBottom:90
      },
      
});

export default SignupScreen;