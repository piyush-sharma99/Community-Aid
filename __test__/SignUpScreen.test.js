import React from "react";
import SignupScreen from "../screens/SignupScreen";
import { firebaseConfig } from "../apikey";
import { cleanup } from "@testing-library/react-native";
import renderer from "react-test-renderer";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);

afterEach(cleanup);

test("SignUp renders correctly", () => {
  const tree = renderer.create(<SignupScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});
