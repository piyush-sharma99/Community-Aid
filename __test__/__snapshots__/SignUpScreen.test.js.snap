// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "height": "100%",
      "width": "100%",
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "100%",
          "width": "100%",
        },
        undefined,
      ]
    }
  />
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableOnAndroid={false}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardSpace={0}
    onScroll={[Function]}
    resetKeyboardSpace={[Function]}
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollIntoView={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    update={[Function]}
    viewIsInsideTabBar={false}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "justifyContent": "center",
              "marginTop": -20,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "#FDFEFE",
              "borderColor": "#fb5b5a",
              "borderRadius": 25,
              "borderWidth": 4,
              "elevation": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginTop": 20,
              "padding": 20,
              "width": "80%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Name..."
            placeholderTextColor="#003f5c"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "black",
                "height": 50,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FDFEFE",
              "borderColor": "#fb5b5a",
              "borderRadius": 25,
              "borderWidth": 4,
              "elevation": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginTop": 20,
              "padding": 20,
              "width": "80%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Number..."
            placeholderTextColor="#003f5c"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "black",
                "height": 50,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FDFEFE",
              "borderColor": "#fb5b5a",
              "borderRadius": 25,
              "borderWidth": 4,
              "elevation": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginTop": 20,
              "padding": 20,
              "width": "80%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Email..."
            placeholderTextColor="#003f5c"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "black",
                "height": 50,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FDFEFE",
              "borderColor": "#fb5b5a",
              "borderRadius": 25,
              "borderWidth": 4,
              "elevation": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginTop": 20,
              "padding": 20,
              "width": "80%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Password..."
            placeholderTextColor="#003f5c"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "black",
                "height": 50,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FDFEFE",
              "borderColor": "#fb5b5a",
              "borderRadius": 25,
              "borderWidth": 4,
              "elevation": 10,
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginTop": 20,
              "padding": 20,
              "width": "80%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Repeat Password..."
            placeholderTextColor="#003f5c"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "color": "black",
                "height": 50,
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
        <View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fb5b5a",
                "borderColor": "#fff",
                "borderRadius": 15,
                "borderWidth": 3,
                "elevation": 10,
                "height": 50,
                "justifyContent": "center",
                "marginBottom": 90,
                "marginTop": 50,
                "opacity": 1,
                "shadowOpacity": 1,
                "width": 200,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "white",
                  "fontSize": 20,
                }
              }
            >
              Sign Up
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
